### PR TITLE
Quick Start: cancel tours when user navigates away

### LINKED ISSUE #10398: Quick Start: Flipping tabs should cancel current tour
Flipping tabs or otherwise leaving the quick start tour's flow should cancel that tour. Currently the notices stick around, which can be annoying and confusing.

### PR DESCRIPTION
**Pull request by nheagy**
Fixes #10398

If the user goes somewhere outside the given tour, it automatically ends.

quick start cancel tour

To test:
 - start a tour
 - ignore it, go somewhere else in the app
 - ensure the notice is dismissed

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/QuickStartTourGuide.swift [hunk 1]
```diff
@@ -342,6 +353,9 @@ internal extension NSNotification.Name {
 @objc
 public enum QuickStartTourElement: Int {
     case noSuchElement
+    case tabFlipped
+    case blogDetails
```
[the remark this answers is not in the document]
Comment (nheagy) [reply WordPress/Classes/ViewRelated/Blog/QuickStartTourGuide.swift]: AH, you're right, that's for the error test that I removed. Will cleanup!

**Comment by nheagy** [comment]
Ok @ScoutHarris how 'bout now?

**Comment by ScoutHarris** [comment]
Thank you sir! That'll do it! :shipit:

**Comment by nheagy** [comment]
Thanks @ScoutHarris!

**Pull request merged**